src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -3470,18 +3471,21 @@ public Long lastsave() {
    * {@link #save() SAVE} and then {@link #quit() QUIT} because other clients may alter the DB data
    * between the two commands.
    * @return Status code reply on error. On success nothing is returned since the server quits and
-   *         the connection is closed.
+   *         the connection is closed. The return type will be void in next major release. Please
+   *         don't use the return.
+   * @deprecated The return type will be void in next major release. Please don't use the return.
    */
   @Override
-  public String shutdown() {
+  @Deprecated
+  public String shutdown() throws JedisException {
     client.shutdown();
-    String status;
     try {
-      status = client.getStatusCodeReply();
-    } catch (JedisException ex) {
-      status = null;
+      String status = client.getStatusCodeReply();
+      throw new JedisException(status);
```

Comment:
@sazzad16 I get that you want to deprecate it, but why do you change the logic?

Reply:
@gkorland I've added throwing exception when there's an error. Other than that, the logic is same. E.g. `null` is returned when everything goes well both in existing code and in new code.